src/main/java/com/google/graphgeckos/dashboard/storage/ParsedBuildbotData.java

```diff
@@ -0,0 +1,107 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.graphgeckos.dashboard.storage;
+
+/**
+ * An immutable container used for providing stripped-down build bot information
+ * to the StorageController. Provides functionality for checking whether the information
+ * is sufficient for an update of an entry. Does not modify or validity check the individual
+ * contents of each field.
+ *
+ * Fields included: 
+ * -commitHash: the commit hash of the revision built by a certain build bot.
+ * -builderName: the name of the build bot which performed the compilation.
+ * -logs: name of the stage and links to the logs for each stage of the compilation.
+ * -status: the final result of the compilation (passed or failed).
```

Comment:
Aren't there also build that never finish?

Reply:
Is it possible for a worker to start compiling a revision and then abandon it?

Reply:
I am not entirely sure if all buildbots can, but I have seen some where this is the case. Some build bots abandon old revisions in favor of a long queue of newer revisions for example. I only want you to be cautious about assuming that the status is always binary, unless you are discarding everything else, though that might make combining data from different bots a challenge.

Reply:
Changed the field to "BuildStatus" type, an enum which can be easily expanded, from PR #36